src/Cake.Curl.Tests/CurlUploadFileTests.cs

```diff
@@ -477,6 +478,76 @@ public void Should_Not_Set_The_RetryConnRefused_Option_As_Argument()
                 // Then
                 Assert.DoesNotContain("--retry-connrefused", result.Args);
             }
+
+            [Theory]
+            [MemberData(nameof(CurlTimeSpanTestData.TimeSpanData), MemberType = typeof(CurlTimeSpanTestData))]
+            public void Should_Set_The_MaxTime_Option_As_Argument(TimeSpan maxTime)
+            {
+                // Given
+                var fixture = new CurlUploadFileFixture
+                {
+                    Settings = { MaxTime = maxTime }
+                };
+
+                // When
+                var result = fixture.Run();
+
+                // Then
+                Assert.Contains(
+                    $"--max-time {maxTime.TotalSeconds.ToString(CultureInfo.CurrentCulture)}",
+                    result.Args);
+            }
+
+            [Fact]
+            public void Should_Not_Set_The_MaxTime_Option_As_Argument()
+            {
+                // Given
+                var fixture = new CurlUploadFileFixture
+                {
+                    Settings = { MaxTime = null }
+                };
+
+                // When
+                var result = fixture.Run();
+
+                // Then
+                Assert.DoesNotContain("--max-time", result.Args);
+            }
+
+            [Theory]
+            [MemberData(nameof(CurlTimeSpanTestData.TimeSpanData), MemberType = typeof(CurlTimeSpanTestData))]
+            public void Should_Set_The_ConnectTimeout_Option_As_Argument(TimeSpan connectionTimeout)
+            {
+                // Given
+                var fixture = new CurlUploadFileFixture
+                {
+                    Settings = { ConnectionTimeout = connectionTimeout }
+                };
+
+                // When
+                var result = fixture.Run();
+
+                // Then
+                Assert.Contains(
+                    $"--connect-timeout {connectionTimeout.TotalSeconds.ToString(CultureInfo.CurrentCulture)}",
```

Comment:
Here, we can simply say `connectionTimeout.TotalSeconds` since the compiler will add the call to `ToString()` when using [string interpolation](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/tokens/interpolated). Also, when invoking `ToString()` on a `Double` the usage of the current culture is implicit.